macOS: use `NSDockTilePlugIn` to update app icons

bo2themax opened the pull request:
* Using MAP prohibited `NSDockTilePlugIn` to update custom icons more reliably. And it also fixes the corner radius issue on older os: #8870
* Changes in AboutWindow cc @jparise 
  * Start cycling with current icon
  * Stop cycling icons when AboutWindow is closed 
  * Add menu to copy icon config

<img width="166" height="63" alt="Xnip2025-12-20_18-40-58" src="https://github.com/user-attachments/assets/52fc1215-909e-49c7-a37a-b7c73eef61f1" />

> [!WARNING]
> Upgrading from `macOS-custom-icon` needs to manually open the app once to update the icon, since this plugin is running under `com.apple.dock.external.extra.arm64`, which has sandbox enabled.
>
> When first upgraded to this commit, a notification about the dock plugin will pop up. The user has to enable this to change the icon smoothly.
> <img width="389" height="159" alt="image" src="https://github.com/user-attachments/assets/a883ac6b-0b4d-4794-8c61-50b60707f6a2" />



Here are the testing results on [Tahoe](https://github.com/user-attachments/assets/e5fc8354-5f5c-4280-805f-88f043ceadca) and [Sequoia](https://github.com/user-attachments/assets/633d9a07-7d9d-4806-8496-82ddaffb8833):

> When you see some pause in the recording, that's when I rebuild or replace the older version with the latest.

This also fixes some issues when changing between different styles, consistency issues, and resetting from others to `official`.


### Developer's Note

This shouldn't affect current CI flow, since this new target is just a bundle not runnable, and I tested with archiving, exporting and signing in Xcode, nothing big changed.


> [!NOTE]
> AI helped me to write the typo ignore-re and proofread my comments

bo2themax committed:
macOS: use `NSDockTilePlugIn` to update app icons

bo2themax committed:
macOS: stop cycling icons when AboutWindow is closed

and start cycling with current icon

mitchellh committed:
moving lots of files, removing unused stuff

mitchellh committed:
macos: rename a bunch of files

mitchellh reviewed (changes requested):
Sorry this took forever to get to. Overall, using an NSDockTilePlugIn sounds great and this implementation is decent. I just shuffled a bunch of stuff around, could you re-review yourself and make sure this all looks/works right?

I think before merge it'd be nice to also clean up the AppIcon stuff so that the enum uses an encoded ColorizedGhosttyIcon rather than that `_`-separated custom encoding we have in this PR. We should be able to make that Coddle if we use a wrapper type around NSColor or something.

bo2themax committed:
macOS: update AppIcon encoding

- make `ColorizedGhosttyIcon` codable
- remove deprecated string encoding introduced in tip

bo2themax committed:
ci: fix linting and delete non-useful tests

mitchellh committed:
macos: move icon codable/equatable to extension

mitchellh committed:
macos: icon tests

mitchellh committed:
macos: various dock tile cleanups
